packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -439,11 +650,46 @@ const Combobox = forwardRef<HTMLDivElement, ComboboxProps>(
           // there's already keyboard handlers to open the listbox on the input element
           onClick={handleInputClick}
         >
+          {multiselect &&
+            selectedValues.map((value, index) => {
+              const refCallback = (elem: HTMLButtonElement | null) => {
+                if (elem) {
+                  pillsRef.current[index] = elem;
+                } else {
+                  pillsRef.current.splice(index, 1);
+                }
+              };
+
+              const handleClick = () =>
+                handleRemove(pillsRef.current[index], value);
+
+              const commonProps = {
+                ref: refCallback,
+                key: value,
+                className: 'Combobox__pill',
+                tabIndex: -1,
+                onClick: handleClick,
+                onKeyDown: handlePillKeyDown
+              };
+
+              return !disabled ? (
+                <TagButton
+                  label=""
+                  value={value || ''}
+                  icon="close"
+                  {...commonProps}
+                />
```

Comment:
One thing that isn't accounted for in the issue description, is what the accessible name for this control should be. We need some way to be able to pass in the accessible name for these pills, so they read something along the lines of "remove {label}" for users of assistive technology.

Otherwise, right now this would just read as "{label}, button" and not really providing a descriptive enough context for AT users.